lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -1491,33 +1491,42 @@ struct LazyLocationListener : public OpBuilder::Listener {
 
 /// Remove the DontTouch annotation and return true, if it exists.
 /// Ignore DontTouch that will apply only to a subfield.
-static bool removeDontTouch(ArrayAttr &annotations) {
+/// Return true if a "circt.nonlocal" annotation found and the checkNonLocal is
+/// set.
+static bool removeDontTouch(ArrayAttr &annotations, bool checkNonLocal = true) {
```

Comment:
Is this parameter every _not_ true? Can we just remove it?

Also, it is confusing that this method is still called `removeDontTouch`, but is really "tell me if a symbol is needed and remove any don't touch annotations". Method name and comment should be updated to reflect this.

Reply:
possible split in to query function and update function.

Reply:
Simplified the code. 
Trying to do multiple tasks here to avoid additional iterations over the entire annotation list.